BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/ShoppingCartWorkflowImpl.java

```diff
@@ -0,0 +1,76 @@
+package barryspeanuts;
+
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.Purchase;
+import barryspeanuts.model.PurchaseItem;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowQueue;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+
+public class ShoppingCartWorkflowImpl implements ShoppingCartWorkflow {
+  private static final Logger logger = Workflow.getLogger(ShoppingCartWorkflowImpl.class);
+
+  boolean exit = false;
+  List<PurchaseItem> purchaseItems = new ArrayList<>();
+  private final WorkflowQueue<Runnable> queue = Workflow.newWorkflowQueue(1024);
+
+  @Override
+  public void startWorkflow() {
+
+    logger.info("Starting Workflow for Barry's Peanuts");
+    while (true) {
+      Workflow.await(() -> exit);
+    }
+  }
+
+  @Override
+  public List<PurchaseItem> queryPurchaseItems() {
+    return this.purchaseItems;
+  }
+
+  @Override
+  public void addItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.add(purchaseItem);
+  }
+
+  @Override
+  public void removeItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.remove(purchaseItem);
+  }
+
+  @Override
+  public void checkOut(Purchase purchase) {
+
+    logger.info("Checking out purchase id {}", purchase.getId());
+  }
+
+  @Override
+  public void pay(Purchase purchase, CreditCard creditCard) {
+    logger.info(
+        "Paying for purchase] id {} using credit card number {}",
```

Comment:
```suggestion
        "Paying for purchase id {} using credit card number {}",
```

Also, don't put the credit card number in the log